quill/common/src/components.rs

```diff
@@ -106,20 +107,73 @@ pub struct CreativeFlying(pub bool);
 
 bincode_component_impl!(CreativeFlying);
 
-type EffectKind = libcraft_effects::PotionEffect;
-#[derive(Clone, Debug, Hash, Serialize, PartialEq, Eq, Deserialize)]
-pub struct PotionEffect {
-    effect: EffectKind,
-    amplifier: u8,
-    duration: u32,
-    particle: bool,
-    ambient: bool, // given from beacon or not.
-    icon: bool,    // show effect icon.
+#[derive(Copy, Clone, Debug, Hash, Serialize, PartialEq, Eq, Deserialize)]
+pub struct PotionApplication {
+    pub amplifier: u8,
+    pub duration: u32,
+    pub particle: bool,
+    pub ambient: bool, // given from beacon or not.
+    pub icon: bool,    // show effect icon.
```

Comment:
Switch the comments to doc comments on the members, consider adding one for `particle` and `duration`.